### PR TITLE
[ML] Transform test fix for testCreateLatestFunction

### LINKED ISSUE #66292: Transform FunctionFactoryTests. testCreateLatestFunction fails
**Build scan**:
https://gradle-enterprise.elastic.co/s/jpxhkvlcqq3ze
**Repro line**:
```
./gradlew ':x-pack:plugin:transform:test' --tests "org.elasticsearch.xpack.transform.transforms.FunctionFactoryTests.testCreateLatestFunction" -Dtests.seed=98F3DDD081D6BC00 -Dtests.security.manager=true -Dtests.locale=ko-KR -Dtests.timezone=Europe/Ljubljana -Druntime.java=11
```
**Reproduces locally?**:
Yes
**Applicable branches**:
master

**Failure excerpt**:
```

org.elasticsearch.xpack.transform.transforms.FunctionFactoryTests > testCreateLatestFunction FAILED |  
-- | --
  | java.lang.IllegalArgumentException: Composite source names must be unique, found duplicates: [O] |  
  | at __randomizedtesting.SeedInfo.seed([98F3DDD081D6BC00:A0E95A82E5AC2FFF]:0) |  
  | at org.elasticsearch.search.aggregations.bucket.composite.CompositeAggregationBuilder.validateSources(CompositeAggregationBuilder.java:202) |  
  | at org.elasticsearch.search.aggregations.bucket.composite.CompositeAggregationBuilder.<init>(CompositeAggregationBuilder.java:79) |  
  | at org.elasticsearch.search.aggregations.AggregationBuilders.composite(AggregationBuilders.java:379) |  
  | at org.elasticsearch.xpack.transform.transforms.latest.Latest.createCompositeAggregation(Latest.java:80) |  
  | at org.elasticsearch.xpack.transform.transforms.latest.Latest.<init>(Latest.java:68) |  
  | at org.elasticsearch.xpack.transform.transforms.FunctionFactory.create(FunctionFactory.java:30) |  
  | at org.elasticsearch.xpack.transform.transforms.FunctionFactoryTests.testCreateLatestFunction(FunctionFactoryTests.java:38)
```

### PR DESCRIPTION
**Pull request by benwtrent**
closes https://github.com/elastic/elasticsearch/issues/66292

### CODE REVIEW

**Comment by benwtrent** [comment]
@elasticmachine update branch

File: x-pack/plugin/core/src/test/java/org/elasticsearch/xpack/core/transform/transforms/latest/LatestConfigTests.java [hunk 1]
```diff
@@ -25,7 +27,13 @@
 public class LatestConfigTests extends AbstractSerializingTransformTestCase<LatestConfig> {
 
     public static LatestConfig randomLatestConfig() {
-        return new LatestConfig(randomList(1, 10, () -> randomAlphaOfLengthBetween(1, 10)), randomAlphaOfLengthBetween(1, 10));
+        return new LatestConfig(
+            new ArrayList<>(
+                new HashSet<>(randomList(1, 10, () -> randomAlphaOfLengthBetween(1, 10))
```
Comment (hendrikmuhs) [inline x-pack/plugin/core/src/test/java/org/elasticsearch/xpack/core/transform/transforms/latest/LatestConfigTests.java:32]: nit, `ESTestCase` has a method `randomUnique`: 
`randomUnique(() -> randomAlphaOfLengthBetween(1, 10), randomIntBetween(1, 10))`

**Review by hendrikmuhs — APPROVED** [review]
LGTM

**Review by przemekwitek — APPROVED** [review]
LGTM

**Comment by hendrikmuhs** [comment]
the same problem exists in 2 other places:
```
org.elasticsearch.client.transform.transforms.latest.hlrc.LatestConfigTests
org.elasticsearch.client.transform.transforms.latest.LatestConfigTests
```

@benwtrent can you fix those, too?

**Pull request merged**